driver/configuration/provider_koanf.go

```diff
@@ -283,12 +289,46 @@ func (v *KoanfProvider) pipelineIsEnabled(prefix, id string) bool {
 	return v.source.Bool(fmt.Sprintf("%s.%s.enabled", prefix, id))
 }
 
+func (v *KoanfProvider) hashPipelineConfig(prefix, id string, override json.RawMessage) (uint64, error) {
+	ts := v.configChangeAt.UnixNano()
+	b := make([]byte, 8)
+	binary.LittleEndian.PutUint64(b, uint64(ts))
+
+	slices := [][]byte{
+		[]byte(prefix),
+		[]byte(id),
+		[]byte(override),
+		b,
+	}
```

Comment:
This cache key will break hot reloading, because the cache key does not respect the config values.